x/axelarnet/handler.go

```diff
@@ -91,3 +98,47 @@ func NewHandler(k keeper.Keeper, n types.Nexus, b types.BankKeeper, a types.Acco
 		return res, nil
 	}
 }
+
+// NewProposalHandler returns the handler for the proposals of the axelarnet module
+func NewProposalHandler(nexusK types.Nexus, accountK types.AccountKeeper) govtypes.Handler {
+	return func(ctx sdk.Context, content govtypes.Content) error {
+		switch c := content.(type) {
+		case *types.CallContractsProposal:
+			for _, contractCall := range c.ContractCalls {
+				sender := nexus.CrossChainAddress{Chain: exported.Axelarnet, Address: accountK.GetModuleAddress(govtypes.ModuleName).String()}
```

Comment:
why all the duplication vs just reusing axelarnet.CallContract? To avoid the chain activation checks?

Reply:
correct. It's also weird to re-use msg_server's logic directly. If anything, a later refactor can only in to extract the common logic into a keeper method.